clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -163,6 +163,15 @@ else if (maxTimestamp == LogEntry.NO_TIMESTAMP)
             return new RecordsInfo(maxTimestamp, compressionType == CompressionType.NONE ? offsetOfMaxTimestamp : lastOffset);
     }
 
+    /**
+     * We set the sequence of the batch just before it is about to be sent out.
+     * @param baseSequence
+     */
+    public void closeWithSequence(int baseSequence) {
+        this.baseSequence = baseSequence;
```

Comment:
The point is to prevent usage errors. What should the behavior be if you use `closeWithSequence()` and then `close()`. How about the other way around?